presto-main/src/main/java/com/facebook/presto/type/LikeFunctions.java

```diff
@@ -94,6 +96,63 @@ public static Regex likePattern(@SqlType("varchar(x)") Slice pattern, @SqlType("
         return likePattern(pattern.toStringUtf8(), getEscapeChar(escape), true);
     }
 
+    public static boolean isLikePattern(Slice pattern, Slice escape)
+    {
+        String stringPattern = pattern.toStringUtf8();
+        if (escape == null) {
+            return stringPattern.contains("%") || stringPattern.contains("_");
+        }
+
+        String stringEscape = escape.toStringUtf8();
+        checkCondition(stringEscape.length() == 1, INVALID_FUNCTION_ARGUMENT, "Escape must be empty or a single character");
+
+        char escapeChar = stringEscape.charAt(0);
+        boolean escaped = false;
+        boolean isLikePattern = false;
+        for (int currentChar : stringPattern.codePoints().toArray()) {
+            if (!escaped && (currentChar == escapeChar)) {
+                escaped = true;
+            }
+            else if (escaped) {
+                checkEscape(currentChar == '%' || currentChar == '_' || currentChar == escapeChar);
+                escaped = false;
+            }
+            else if ((currentChar == '%') || (currentChar == '_')) {
+                isLikePattern = true;
```

Comment:
Probably this has already been reviewed (sorry, I couldn't read all the history), but for me it's quite surprising there is no return nor break here. Since this `isLikePattern` method checks whether LIKE pattern is really a pattern, once we see '%' we need to look no further.

I understand there must be a place in the code which checks LIKE pattern for wellformedness, but I would expect this place to be called `checkLikePattern`. I.e. the fact `boolean isLikePattern(..)`  has a side effect would be surprising to me.